pandas/core/groupby/groupby.py

```diff
@@ -2235,8 +2235,11 @@ def _get_cythonized_result(
         grouper = self.grouper
 
         labels, _, ngroups = grouper.group_info
-        output: Dict[base.OutputKey, np.ndarray] = {}
+        output: Dict[base.OutputKey, np.ndarray, str:str] = {}
         base_func = getattr(libgroupby, how)
+        obj = self._selected_obj
+        if isinstance(obj, DataFrame):
```

Comment:
what are you doing here? this is very odd

Reply:
The function call ```self._wrap_aggregated_output``` or ```self._wrap_transformed_output(output)``` convert the array containing the results in a DataFrame (or Series, but not important in our case). The name of the index is already lost at this point, so I added the name to the output dictionary. This is not relevant, if the result is a Series, so I check, if we have a DataFrame.

My first idea was to check if the original object was from the type ```DataFrameGroupBy```, but I could not perform this check without importing ```DataFrameGroupBy``` during runtime. To avoid this, I used the method you see above.

Does this answer your question?